### PR TITLE
clang-build-analyzer 1.4.0 (new formula)

### PR DESCRIPTION
**Pull request by ktf**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by SMillerDev** [comment]
```
  ==> Testing clang-build-analyzer
  ==> c++ -c -ftime-trace /tmp/clang-build-analyzer-test-20230123-5633-12u1prm/test.cxx
  c++: error: unrecognized command-line option ‘-ftime-trace’
```

**Comment by ktf** [comment]
‘-ftime-trace’ is there only for clang. I made the recipe macOS only.

**Comment by ktf** [comment]
Now it passes all the tests.

File: Formula/clang-build-analyzer.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
+class ClangBuildAnalyzer < Formula
+  desc "Tool to analyze compilation time"
+  homepage "https://github.com/aras-p/ClangBuildAnalyzer"
+  url "https://github.com/aras-p/ClangBuildAnalyzer/archive/v1.4.0.tar.gz"
+  sha256 "dae8e7838145a72c01c397c3998d9f6801fc4dc819d552010d702cab7dede530"
+  license "Unlicense"
+  head "https://github.com/aras-p/ClangBuildAnalyzer.git", branch: "main"
+
+  depends_on "cmake" => :build
+  depends_on :macos
```
Comment (carlocab) [inline Formula/clang-build-analyzer.rb:10]: Why not
```suggestion
  uses_from_macos "llvm"
```
? It seems weird to make this depend on macOS when what it needs is Clang and not macOS itself.

File: Formula/clang-build-analyzer.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  test do
+    # `test do` will create, run in and delete a temporary directory.
+    #
+    # This test will fail and we won't accept that! For Homebrew/homebrew-core
+    # this will need to be a test that verifies the functionality of the
+    # software. Run the test with `brew test clang-build-analyzer`. Options passed
+    # to `brew install` such as `--HEAD` also need to be provided to `brew test`.
+    #
+    # The installed folder is not in the path, so use the entire path to any
+    # executables being tested: `system "#{bin}/program", "do", "something"`.
```
Comment (carlocab) [inline Formula/clang-build-analyzer.rb:27]:
```suggestion
```

File: Formula/clang-build-analyzer.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    (testpath/"test.cxx").write <<~EOS
+      int main() {}
+    EOS
+    system "c++", "-c", "-ftime-trace", testpath/"test.cxx"
```
Comment (carlocab) [inline Formula/clang-build-analyzer.rb:31]:
```suggestion
    ENV.clang
    system ENV.cxx, "-c", "-ftime-trace", testpath/"test.cxx"
```

File: Formula/clang-build-analyzer.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    system "ClangBuildAnalyzer", "--all", testpath, "test.db"
+    system "ClangBuildAnalyzer", "--analyze", "test.db"
```
Comment (carlocab) [inline Formula/clang-build-analyzer.rb:33]:
```suggestion
    system bin/"ClangBuildAnalyzer", "--all", testpath, "test.db"
    system bin/"ClangBuildAnalyzer", "--analyze", "test.db"
```

File: Formula/clang-build-analyzer.rb [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  sha256 "dae8e7838145a72c01c397c3998d9f6801fc4dc819d552010d702cab7dede530"
+  license "Unlicense"
+  head "https://github.com/aras-p/ClangBuildAnalyzer.git", branch: "main"
+
+  depends_on "cmake" => :build
+  depends_on :macos
+
+  def install
+    system "cmake", "-S", ".", "-B", "build", *std_cmake_args
+    system "make", "-C", "build"
```
Comment (carlocab) [inline Formula/clang-build-analyzer.rb:14]:
```suggestion
    system "cmake", "--build", "build"
```

File: Formula/clang-build-analyzer.rb [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    bin.install ["build/ClangBuildAnalyzer"]
```
Comment (carlocab) [inline Formula/clang-build-analyzer.rb:15]: Why aren't we using an `install` target?

Comment (ktf) [reply Formula/clang-build-analyzer.rb]: Fixed

**Comment by ktf** [comment]
I have applied all the suggested changes and squashed everything again in one single commit.

**Comment by ktf** [comment]
All the tests green again.

File: Formula/clang-build-analyzer.rb [hunk 7]
```diff
@@ -0,0 +1,26 @@
+class ClangBuildAnalyzer < Formula
+  desc "Tool to analyze compilation time"
+  homepage "https://github.com/aras-p/ClangBuildAnalyzer"
+  url "https://github.com/aras-p/ClangBuildAnalyzer/archive/v1.4.0.tar.gz"
+  sha256 "dae8e7838145a72c01c397c3998d9f6801fc4dc819d552010d702cab7dede530"
+  license "Unlicense"
```
[the remark this answers is not in the document]
Comment (ktf) [reply Formula/clang-build-analyzer.rb:6]: What should I do in that case? Have all the licenses as a list?

[the remark this answers is not in the document]
Comment (carlocab) [reply Formula/clang-build-analyzer.rb:6]: Yes. You can use the `all_of:` syntax to specify an array of licenses.

**Review by carlocab — APPROVED** [review]
LGTM. Thanks!

**Comment by ktf** [comment]
Thank you for the detailed review.

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
@ktf, thanks for your contribution to Homebrew! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Pull request closed without merging**